### ISSUE TITLE
new Bitmask() or Bitmask::from doesn't take a bitmask

### ISSUE THREAD

**Issue by Externaluse** [cruxinator/php-bitmask#17]
Please forgive me if this is stupid.
I had expected to be able to instantiate a BitMask from a compiled integer, ie so that `new Bitmask(3)` would later allow me to check if both the Enum with the value of 1 and the one with the value of 2 would be set. This doesn't appear to work.

    namespace app\Bitmasks;
    use Cruxinator\BitMask\BitMask;
    class TestFlags extends BitMask
    {
        const GROUPA = 1 << 0;
        const GROUPB = 1 << 1;
        const GROUPC = 1 << 2;
    }`
Now I was expecting to be able to do this:

    $x = new TestFlags(3);
    $x->isGROUPA(); // expected true
    $x->isGROUPB(); // expected true
    $x->isGROUPC(); // expected false

Instead of my expectation I receive `Uncaught UnexpectedValueException: Value '3' is not part of the enum app\Bitmasks\TestFlags`

Am I missing something obvious?

**Comment by Externaluse** [comment]
I've helped myself for the moment by extending BitMask to include a `setValue(int $value)` method that writes directly to `$this->value`. Not sure if it was intended this way though.

    abstract class BitMask extends \Cruxinator\BitMask\BitMask
    {
        /**
         * Required because ::from and __construct would use enums constructor, not allowing a combined mask to be stored
         * @param int $value
         * @return BitMask
         */
        public function setValue(int $value): BitMask
        {
            $this->value = $value;
            return $this;
        }
    }

**Comment by c-harris** [comment]
@Externaluse good catch, ill address that one first chance i get. thanks for letting me know. i didn't actually think anyone was using it so i havent ket an eye on it.

**Comment by Tsaukpaetra** [comment]
+1 on this fix. Now I need to figure out why `getKey()` isn't working as expected...
Edit: Oh,. that's actually working, it's the magic functions that aren't. Will raise another issue if I can't figure it out...

**Comment by c-harris** [comment]
@Externaluse sorry i think that was a new version of the underlying enum that I had to limit.
